src/Blocks/BlockSponge.h

```diff
@@ -0,0 +1,125 @@
+
+#pragma once
+
+#include "BlockHandler.h"
+
+
+
+
+
+class cBlockSpongeHandler :
+	public cBlockHandler
+{
+public:
+	cBlockSpongeHandler(BLOCKTYPE a_BlockType):
+		cBlockHandler(a_BlockType)
+	{
+	}
+
+
+
+
+
+	virtual void Check(cChunkInterface & a_ChunkInterface, cBlockPluginInterface & a_PluginInterface, int a_RelX, int a_RelY, int a_RelZ, cChunk & a_Chunk) override
+	{
+		if (GetSoaked(Vector3i(a_RelX, a_RelY, a_RelZ), a_Chunk))
+		{
+			return;
+		}
+		cBlockHandler::Check(a_ChunkInterface, a_PluginInterface, a_RelX, a_RelY, a_RelZ, a_Chunk);
+	}
+
+
+
+
+
+	/** Check blocks above and around to see if they are water.
+		If one is, soak this sponge and remove nearby water.
+		Returns TRUE if the block was changed. */
+	bool GetSoaked(Vector3i a_Rel, cChunk & a_Chunk)
+	{
+		static const std::array<Vector3i, 6> WaterCheck
+		{
+			{
+				{ 1,  0,  0},
+				{-1,  0,  0},
+				{ 0,  0,  1},
+				{ 0,  0, -1},
+				{ 0,  1,  0},
+				{ 0, -1,  0},
+			}
+		};
+		struct sSeed
+		{
+			sSeed(Vector3i pos, int d)
+			{
+				m_Pos = pos;
+				m_Depth = d;
+			}
+			Vector3i m_Pos;
+			int m_Depth;
+		};
+		NIBBLETYPE TargetMeta = a_Chunk.GetMeta(a_Rel.x, a_Rel.y, a_Rel.z);
+		if (TargetMeta != E_META_SPONGE_DRY)
+		{
+			return false;
+		}
+
+		bool ShouldSoak = std::any_of(WaterCheck.cbegin(), WaterCheck.cend(), [a_Rel, & a_Chunk](Vector3i a_Offset)
+			{
+				return IsWet(a_Rel + a_Offset, a_Chunk);
+			}
+		);
+
+		if (! ShouldSoak)
+		{
+			return false;
+		}
```

Comment:
Add comments about what the code is doing, in groups.

Group lines 62 to 77 into a single block, with comment along the lines of `Check if this is a dry sponge next to a water block`